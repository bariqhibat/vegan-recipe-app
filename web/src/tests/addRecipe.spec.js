import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import ApolloMockedProvider from "./utils/provider";

import AddRecipe from "../routes/addRecipe";

const jestMock = jest.fn();

describe("Add Recipe", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("Accessability", () => {
    test("Snapshot", async () => {
      const { asFragment } = render(
        <ApolloMockedProvider>
          <Router>
            <AddRecipe jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      await waitFor(() => expect(asFragment()).toMatchSnapshot());
    });
    test("Mock", async () => {
      render(
        <ApolloMockedProvider showErrors={false}>
          <Router>
            <AddRecipe jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      expect(jestMock).toHaveBeenCalled();
      await waitFor(() =>
        expect(jestMock).toHaveBeenLastCalledWith("addRecipe rendered!")
      );
    });
  });
});
