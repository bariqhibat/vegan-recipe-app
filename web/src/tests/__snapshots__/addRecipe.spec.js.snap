// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipe Accessability Snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding-top: 2rem; display: flex; justify-content: center;"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-card ant-card-bordered"
          style="padding: 2rem 5rem; max-width: 50rem; min-width: 40rem; width: 100%;"
        >
          <div
            class="ant-card-body"
          >
            <h2>
              Create a recipe
            </h2>
            <div
              style="text-align: center; align-content: center;"
            >
              <form
                class="ant-form ant-form-vertical"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="title"
                      title="Title"
                    >
                      Title
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="title"
                          placeholder="Chinese eggplant with spicy szechuan sauce"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="file"
                      title="Image"
                    >
                      Image
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-upload-picture-card-wrapper avatar-uploader"
                        >
                          <div
                            class="ant-upload ant-upload-select ant-upload-select-picture-card"
                          >
                            <span
                              class="ant-upload"
                              role="button"
                              tabindex="0"
                            >
                              <input
                                accept=""
                                id="file"
                                style="display: none;"
                                type="file"
                              />
                              <div>
                                <span
                                  aria-label="plus"
                                  class="anticon anticon-plus"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="plus"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <defs>
                                      <style />
                                    </defs>
                                    <path
                                      d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                    />
                                    <path
                                      d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                    />
                                  </svg>
                                </span>
                                <p>
                                  Upload
                                </p>
                              </div>
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-dashed ant-btn-block"
                            type="button"
                          >
                            <span
                              aria-label="plus"
                              class="anticon anticon-plus"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="plus"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <defs>
                                  <style />
                                </defs>
                                <path
                                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                />
                                <path
                                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                />
                              </svg>
                            </span>
                            <span>
                              Click here to add ingredient
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-dashed ant-btn-block"
                            type="button"
                          >
                            <span
                              aria-label="plus"
                              class="anticon anticon-plus"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="plus"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <defs>
                                  <style />
                                </defs>
                                <path
                                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                />
                                <path
                                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                />
                              </svg>
                            </span>
                            <span>
                              Click here to add instruction
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <button
                  class="ant-btn"
                  type="submit"
                >
                  <span>
                    Submit
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
